src/Dialogs/Settings/Panels/WeatherConfigPanel.cpp

```diff
@@ -49,13 +58,31 @@ class WeatherConfigPanel final
   /* methods from Widget */
   virtual void Prepare(ContainerWindow &parent, const PixelRect &rc) override;
   virtual bool Save(bool &changed) override;
+
+  private:
+  /* methods from DataFieldListener */
+  virtual void OnModified(DataField &df) override {};
```

Comment:
What's the point of implementing the interface if you leave the method empty?

Reply:
When I remove it, I have following error. It makes DataFieldListener::OnModified() empty. But I am not clear. I am glad if someone tell me why.

sc/Dialogs/Settings/Panels/WeatherConfigPanel.cpp: In function ‘Widget* CreateWeatherConfigPanel()’:
src/Dialogs/Settings/Panels/WeatherConfigPanel.cpp:144:33: error: invalid new-expression of abstract class type ‘WeatherConfigPanel’
   return new WeatherConfigPanel();
                                 ^
src/Dialogs/Settings/Panels/WeatherConfigPanel.cpp:51:7: note:   because the following virtual functions are pure within ‘WeatherConfigPanel’:
 class WeatherConfigPanel final
       ^~~~~~~~~~~~~~~~~~
In file included from src/Dialogs/Settings/Panels/WeatherConfigPanel.cpp:33:0:
./src/Form/DataField/Listener.hpp:31:16: note: 	virtual void DataFieldListener::OnModified(DataField&)
   virtual void OnModified(DataField &df) = 0;
                ^~~~~~~~~~

Reply:
I did not ask about the method. I asked about the interface.